ext/mbstring/tests/mb_str_split_jp.phpt

```diff
@@ -0,0 +1,52 @@
+--TEST--
+mb_str_split() tests for the japanese language
+--SKIPIF--
+<?php extension_loaded('mbstring') or die('skip mbstring not available'); ?>
+--INI--
+output_handler=
+mbstring.func_overload=0
+--FILE--
+<?php
+ini_set('include_path','.');
+include_once('common.inc');
+
+$string = "日本";             /* 2 chars */
+$len = 2;
+$charset = [
+    "BIG-5", "EUC-JP", "ISO-2022-JP",
+     "SJIS", "UTF-16BE", "UTF-16LE", "UTF-32BE",
+      "UTF-32LE", "UTF-8"
+];
+
+
+foreach($charset as $cs){
+    $enc = mb_convert_encoding($string, $cs, "UTF-8");
+    $split = mb_str_split($enc, 1, $cs);
+
+    /* check length */
+    for($i = 1; $i <= $len; ++$i){
+        $ceil = ceil($len / $i);
+        if($ceil != count(mb_str_split($enc,$i,$cs))) echo "ERROR";
+    }
+
+    /* check content */
+    echo "$cs: ";
+    for($i = 0; $i < $len; ++$i){
+        echo unpack("H*", $split[$i])[1];
```

Comment:
Could we have a space between each so we can see the actual split?

Reply:
good idea.